### PR TITLE
uploader.js: upload m4a as audio/mp4 | link.js: embed audio/mp4, x-flac, and x-m4a

### PR DESCRIPTION
**Pull request by xnaas**
Trying to follow the MDN, it says that `.m4a` should be set to `audio/mp4`.

This PR makes it so that an `.m4a` file uploaded through The Lounge has the mime type `audio/mp4` set. This also ensures that files being served with `audio/mp4` are embedded by The Lounge.

The only sample `.m4a` files I could find to test with that wasn't something I already have or uploaded were:

* https://dl.espressif.com/dl/audio/ff-16b-2c-44100hz.m4a
* https://dl.espressif.com/dl/audio/ff-16b-1c-44100hz.m4a

but these set `audio/mpeg`, which TL already handles. 😅 It would seem that `.m4a`s are quite rare to just be openly hosted around.

Edit: I found https://filesamples.com/samples/audio/m4a/Symphony%20No.6%20(1st%20movement).m4a which seems to serve as `audio/x-m4a`.

---

Source: https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Containers#browser_compatibility

Relevant previous PR: #4210.

### CODE REVIEW

File: src/plugins/uploader.js
Comment (itsjohncs) [inline src/plugins/uploader.js:118]: This is done because the `file-type` module returns `audio/x-flac` rather than `audio/flac`. This sets it to `audio/flac` instead, which I think is generally what you're trying to do.

Removing this seems like it's counterproductive. Unless you noticed that the newer versions of `file-type` return `audio/flac` now?

File: src/plugins/uploader.js
Comment (itsjohncs) [inline src/plugins/uploader.js:24]: These are compared to the return values of `file-path`, so this should break file-type detection unless `file-path` has changed what mime types it returns for these file types?

**Review by itsjohncs — CHANGES_REQUESTED** [review]
I only took a close look at the `uploader.js` changes, but I'm not sure I understand what's going on. If these changes are working as you expect, could you share your test procedures? (ie: steps to follow, expected result before change, expected result after change)

File: src/plugins/uploader.js
Comment (xnaas) [reply src/plugins/uploader.js:118]: I wonder if this `file-type` plugin is the one responsible for #4429? 🤔

**Comment by xnaas** [comment]
I apologize for not mentioning this in the original PR message.

The goal of this PR is for The Lounge to embed both uploaded `.m4a` files (which get `audio/x-m4a`) as well as `.m4a` files that are served with `audio/mp4` from webservers.

The current situation is that The Lounge will only embed a `.m4a` that you uploaded directly to it or if a webserver is configured to add the `audio/x-m4a` mime type to a `.m4a` file instead of `audio/mp4`.

**Comment by xnaas** [comment]
Without this PR: bad

With this PR: good

**Comment by xnaas** [comment]
I built [this container image](https://github.com/xnaas/docker-thelounge/pkgs/container/thelounge/14838682) off of https://github.com/thelounge/thelounge/pull/4470/commits/daaf0c3e9a843608ef4ac03a6cad00bcff099544 to make testing easy for anyone already versed in Docker.

**Comment by xnaas** [comment]
Also, thank you for the feedback bookworm. :)

**Review by itsjohncs — APPROVED** [review]
Looks good to me. Didn't test it myself but took a close look at the code and feel reasonably confident this'll do what's advertised. Thank you! I'll merge this tomorrow if no other maintainers chime in (and if I don't forget 😅).

**Review by MaxLeiter — APPROVED** [review]

**Comment by MaxLeiter** [comment]
Thanks @xnaas!

**Pull request merged**
